### PR TITLE
Use FLoat32Array in XRRigidTransform

### LINKED ISSUE #31047: WebIDL: use FLoat32Array in XRRigidTransform
Depends on https://github.com/servo/servo/pull/30990

How to do it:

  1.  In [CodegenRust.py](https://github.com/servo/servo/blob/main/components/script/dom/bindings/codegen/CodegenRust.py) remove XRRigidTransform from the list in todo_switch_float_32.
  2. Replace [`matrix: Heap<*mut JSObject>`](https://github.com/servo/servo/blob/355567186400d0531657d0eedda7625de9d0e93a/components/script/dom/xrrigidtransform.rs#L34C5-L34C32) with `HeapFloat32Array`. 
  3. Replace [`Matrix`](https://github.com/servo/servo/blob/355567186400d0531657d0eedda7625de9d0e93a/components/script/dom/xrrigidtransform.rs#L137) to return a `Float32Array`, by using `set_data` and doing the same as in [AudioBuffer::GetChannelData](https://github.com/servo/servo/pull/30990/files#diff-88b03ac9bcc2650883607d41bc374d7f3976567cfbee702cc124f14ce2fbc79bR231)
  
Any questions, ask here or in https://servo.zulipchat.com/

### PR DESCRIPTION
**Pull request by Taym95**
---

- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] These changes fix #31047

- [x] These changes do not require tests because doesn't change behavior

### CODE REVIEW

File: components/script/dom/xrrigidtransform.rs [hunk 1]
```diff
@@ -134,15 +134,11 @@ impl XRRigidTransformMethods for XRRigidTransform {
...
-        if self.matrix.get().is_null() {
-            let cx = GlobalScope::get_cx();
-            // According to the spec all matrices are column-major,
-            // however euclid uses row vectors so we use .to_array()
-            let arr = self.transform.to_transform().to_array();
-            create_typed_array(cx, &arr, &self.matrix);
+    fn Matrix(&self, _cx: JSContext) -> Float32Array {
+        match self.matrix.get_internal() {
+            Ok(matrix) => matrix,
+            Err(_) => todo!(),
```
Comment (Taym95) [inline components/script/dom/xrrigidtransform.rs:140]: @gterzian what should we return in this case?

Comment (gterzian) [reply components/script/dom/xrrigidtransform.rs:140]: What you want to do is: in the `!self.matrix.is_initialized` branch, use `set_data` to initialize the float 32 array to the data of `self.transform`, and only then call `self.matrix.get_internal().expect`

**Review by gterzian — CHANGES_REQUESTED** [review]
Thanks for the contribution! Clarifying things a bit...

File: components/script/dom/xrrigidtransform.rs [hunk 2]
```diff
@@ -134,15 +132,16 @@ impl XRRigidTransformMethods for XRRigidTransform {
...
-            let cx = GlobalScope::get_cx();
-            // According to the spec all matrices are column-major,
-            // however euclid uses row vectors so we use .to_array()
-            let arr = self.transform.to_transform().to_array();
-            create_typed_array(cx, &arr, &self.matrix);
+    fn Matrix(&self, _cx: JSContext) -> Float32Array {
+        if !self.matrix.is_initialized() {
+            self.matrix
+                .set_data(_cx, &self.transform.to_transform().to_array())
+                .expect("failed to set projection matrix")
```
Comment (gterzian) [inline components/script/dom/xrrigidtransform.rs:139]: "Failed to set on data on transform's internal matrix."

File: components/script/dom/xrrigidtransform.rs [hunk 3]
```diff
@@ -134,15 +132,16 @@ impl XRRigidTransformMethods for XRRigidTransform {
...
         }
-        NonNull::new(self.matrix.get()).unwrap()
+
+        self.matrix
+            .get_internal()
+            .expect("failed to get projection matrix")
```
Comment (gterzian) [inline components/script/dom/xrrigidtransform.rs:144]: "Failed to get transform's internal matrix."

**Review by gterzian — APPROVED** [review]
LGTM, with a couple of nits

**Review by gterzian — APPROVED** [review]
LGTM, thanks for the contrib, we can merge this one after https://github.com/servo/servo/pull/31077 has, which will probably require fixing a merge conflict first...

**Pull request merged**